src/test/documentation_examples.rs

```diff
@@ -0,0 +1,1385 @@
+            assert_eq!(status, "P");
+
+            assert!(doc.contains_key("lastModified"));
+        }
+    );
+
+    // Start Example 54
+    collection
+        .replace_one(
+            doc! { "item": "paper" },
+            doc! {
+                "item": "paper",
+                "instock": [
+                    {
+                        "warehouse": "A",
+                        "qty": 60,
+                    },
+                    {
+                        "warehouse": "B",
+                        "qty": 40,
+                    },
+                ],
+            },
+            None,
+        )
+        .await
+        .unwrap();
+    // End Example 54
+
+    run_on_each_doc!(
+        collection
+            .find(doc! { "item": "paper" }, None,)
+            .await
+            .unwrap(),
+        doc,
+        {
+            assert_eq!(doc.len(), 3);
+            assert!(doc.contains_key("_id"));
+            assert!(doc.contains_key("item"));
+            assert!(doc.contains_key("instock"));
+
+            let instock = doc.get_array("instock").unwrap();
+            assert_eq!(instock.len(), 2);
+        }
+    );
+
+    Ok(())
+}
+
+async fn delete_examples(collection: &Collection) -> Result<()> {
+    collection.drop(None).await.unwrap();
+
+    // Start Example 55
+    let docs = vec![
+        doc! {
+            "item": "journal",
+            "qty": 25,
+            "size": {
+                "h": 14,
+                "w": 21,
+                "uom": "cm",
+            },
+            "status": "A",
+        },
+        doc! {
+            "item": "notebook",
+            "qty": 50,
+            "size": {
+                "h": 8.5,
+                "w": 11,
+                "uom": "in",
+            },
+            "status": "P",
+        },
+        doc! {
+            "item": "paper",
+            "qty": 100,
+            "size": {
+                "h": 8.5,
+                "w": 11,
+                "uom": "in",
+            },
+            "status": "D",
+        },
+        doc! {
+            "item": "planner",
+            "qty": 75,
+            "size": {
+                "h": 22.85,
+                "w": 30,
+                "uom": "cm",
+            },
+            "status": "D",
+        },
+        doc! {
+            "item": "postcard",
+            "qty": 45,
+            "size": {
+                "h": 10,
+                "w": 15.25,
+                "uom": "cm",
+            },
+            "status": "A",
+        },
+    ];
+
+    collection.insert_many(docs, None).await?;
+    // End Example 55
+
+    assert_coll_count!(collection, 5);
+
+    // Start Example 57
+    collection.delete_many(doc! { "status": "A" }, None).await?;
+    // End Example 57
+
+    assert_coll_count!(collection, 3);
+
+    // Start Example 58
+    collection.delete_one(doc! { "status": "D" }, None).await?;
+    // End Example 58
+
+    assert_coll_count!(collection, 2);
+
+    // Start Example 56
```

Comment:
This example is explicitly out of order because deleting all the documents in the collection makes trying to delete a subset of them superfluous. See the discussion on the DRIVERS ticket for verification that the docs team doesn't require the examples to be in order.